jira-config/jira-config-api/src/main/java/org/xwiki/contrib/jira/config/JIRAAuthenticator.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.contrib.jira.config;
+
+import org.apache.hc.client5.http.ContextBuilder;
+import org.apache.hc.client5.http.classic.methods.HttpUriRequest;
+import org.apache.hc.core5.http.HttpHost;
+import org.xwiki.component.annotation.Role;
+
+import com.atlassian.jira.rest.client.api.AuthenticationHandler;
+
+/**
+ * JIRA authenticator, to send authenticated request to the target JIRA server.
+ *
+ * @version $Id$
+ * @since 11.3.0
```

Comment:
Wrong since: that would be a 10.3.0 or a 11.0.0 depending if we're going to a major release or not.

Reply:
Given your changes and the possible breaking APIs, it feels like a next major version so 11.0.0